### ISSUE TITLE
Flavors: launch desktop session

### ISSUE THREAD

**Issue by d-loose** [canonical/ubuntu-desktop-provision#299]
Split off from #298.

For flavors that don't use GDM, we need to implement services that take the user to the desktop session after first boot init, as well. LightDM and SDDM are probably the most important to consider. Are there any others?

In general, I'm wondering what the first boot experience looks like for other flavors in the case where no user account has been created yet. Do they all support a minimal desktop session that can run an init wizard, like gnome-initial-setup?

**Comment by spydon** [comment]
> Do they all support a minimal desktop session that can run an init wizard, like gnome-initial-setup?

Isn't that up to the DE/WM?

**Comment by eeickmeyer** [comment]
> For flavors that don't use GDM, we need to implement services that take the user to the desktop session after first boot init, as well. LightDM and SDDM are probably the most important to consider. Are there any others?

That shouldn't be for you to worry about. The flavor maintainers can worry about that part themselves. For Ubuntu Studio, for instance, I can script a separate default to remove the panel and desktop icons in plasma the same way Ubuntu Desktop does currently until the first launch of `ubuntu-desktop-installer` is closed initially, but by using some Javascript calls that Plasma uses. Same concept, different implementation.

Most flavor leads/maintainers are going to want to boot to their live environments and just leave it up to the user to double-click on the "Install RELEASE" icon on the desktop, so I don't think this is even a factor.

**Comment by eeickmeyer** [comment]
> Isn't that up to the DE/WM?

Exactly. @spydon . Historically, Ubiquity would just use whatever WM was installed/available. For Xfce, that was `xfwm4`. For Plasma, that's `kwin`. For LXQt, that's `openbox`, believe it or not. In `xfwm` and `kwin`'s case, they had difficulty rendering a background, so ubiquity would also take advantage of `feh` to render the background if available.

So, if it **can** be done, that would be ideal.

**Comment by jbicha** [comment]
I don't believe any Ubuntu flavor has ever offered a minimal first boot experience like gnome-initial-setup. Even Ubuntu Desktop has only done this until now for OEM mode.

Several but not all desktop flavors offer a welcome app similar to how Ubuntu has used gnome-initial-setup the past few years. (This is a different mode than what I believe is planned for 24.04 LTS.) I don't think this needs any special handling from the installer at all. Ubiquity and Calamares did not need to handle the welcome apps.
